ksql-parser/src/main/java/io/confluent/ksql/schema/ksql/SqlValueCoercer.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.schema.ksql;
+
+import java.util.Optional;
+
+/**
+ * Coerces values to {@link SqlType SQL types}.
+ */
+public interface SqlValueCoercer {
+
+  /**
+   * Coerce the supplied {@code value} to the supplied {@code sqlType}.
+   *
+   * <p>Complex SQL types are not supported, (yet).
+   *
+   * @param value the value to try to coerce.
+   * @param targetSqlType the target SQL type.
+   * @return the coerced value if the value could be coerced, {@link Optional#empty()} otherwise.
+   */
+  Optional<Object> coerce(Object value, SqlType targetSqlType);
```

Comment:
might be nice to just `<T,S> Optional<S> coerce(T value, SqlType target);` to avoid use of `Object` everywhere

Reply:
I'm dubious at to the value of making the return value typed, as I reckon the generics will never be used, but have added `<T>`.

I've not added `<S>` as there would be zero benefit. Erasure would return the type anyway, and it would not allow for any type checking or anything that can't be achieved with `Object`.